File: src/main/java/io/github/electricpoultry/andonium/util/ModTags.java
package io.github.electricpoultry.andonium.util;

import net.minecraft.block.Block;
import net.minecraft.item.Item;
import net.minecraft.registry.RegistryKeys;
import net.minecraft.registry.tag.TagKey;
import net.minecraft.util.Identifier;

import static io.github.electricpoultry.andonium.Andonium.MOD_ID;

public class ModTags {
    public static class Blocks {
        public static final TagKey<Block> NEEDS_ANDONIUM_TOOL = createTag("needs_andonium_tool");
        public static final TagKey<Block> INCORRECT_FOR_ANDONIUM_TOOL = createTag("incorrect_for_andonium_tool");

        private static TagKey<Block> createTag(String name) {
            return TagKey.of(RegistryKeys.BLOCK, Identifier.of(MOD_ID, name));
        }
    }

    public static class Items {
        public static final TagKey<Item> TRANSFORMABLE_ITEMS = createTag("transformable_items");

        private static TagKey<Item> createTag(String name) {
            return TagKey.of(RegistryKeys.ITEM, Identifier.of(MOD_ID, name));
        }
    }
}

File: src/main/java/io/github/electricpoultry/andonium/util/ModTags.kt
package io.github.electricpoultry.andonium.util

import io.github.electricpoultry.andonium.Andonium
import net.minecraft.resources.ResourceLocation
import net.minecraft.tags.BlockTags
import net.minecraft.tags.ItemTags
import net.minecraft.tags.TagKey
import net.minecraft.world.item.Item
import net.minecraft.world.level.block.Block

class ModTags {
    object Blocks {
        val NEEDS_ANDONIUM_TOOL: TagKey<Block> = createTag("needs_andonium_tool")
        val INCORRECT_FOR_ANDONIUM_TOOL: TagKey<Block> = createTag("incorrect_for_andonium_tool")

        private fun createTag(name: String): TagKey<Block> {
            return BlockTags.create(ResourceLocation.fromNamespaceAndPath(Andonium.MOD_ID, name))
        }
    }

    object Items {
        val FAKE_TAG_ITEMS: TagKey<Item> = createTag("fake_tag_items")
        val ANDONIUM_REPAIR: TagKey<Item> = createTag("andonium_repair")

        private fun createTag(name: String): TagKey<Item> {
            return ItemTags.create(ResourceLocation.fromNamespaceAndPath(Andonium.MOD_ID, name))
        }
    }
}
